superset-frontend/src/explore/components/ControlPanelsContainer.jsx

```diff
@@ -159,7 +159,11 @@ class ControlPanelsContainer extends React.Component {
                 // When the item is a React element
                 return controlItem;
               }
-              if (controlItem.name && controlItem.config) {
+              if (
+                controlItem.name &&
+                controlItem.config &&
+                controlItem.name !== 'datasource'
```

Comment:
The datasource control needs to be passed in and filtered out of the view because it is tied to the logic of fetching the chart data through form data.